service/handlers.go

```diff
@@ -0,0 +1,185 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/gorilla/mux"
+	"github.com/jackc/pgx/v4"
+	"github.com/mailgun/service/models"
+	"net/http"
+	"os"
+)
+
+const (
+	DomainThreshold  = 1000
+	BouncedThreshold = 1
+)
+
+type BaseHandler struct {
+	db *pgx.Conn
+}
+
+func NewBaseHandler(db *pgx.Conn) *BaseHandler {
+	return &BaseHandler{
+		db: db,
+	}
+}
+
+func (h *BaseHandler) DeliveredHandler() http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+
+		errResp, domainName := getDomain(r)
```

Comment:
If errResp is always false until it is set below (based on nil-check of `err`), is this variable needed at all?

Reply:
You are correct. this corresponds with the previous bug. Thank you